app/bundles/LeadBundle/Form/Type/MergeType.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * @package     Mautic
+ * @copyright   2014 Mautic Contributors. All rights reserved.
+ * @author      Mautic
+ * @link        http://mautic.org
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Form\Type;
+
+use Mautic\CoreBundle\Factory\MauticFactory;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/**
+ * Class MergeType
+ *
+ * @package Mautic\LeadBundle\Form\Type
+ */
+class MergeType extends AbstractType
+{
+
+    /**
+     * @var MauticFactory
+     */
+    private $factory;
+
+    public function __construct(MauticFactory $factory)
+    {
+        $this->factory = $factory;
+    }
+
+    /**
+     * @param FormBuilderInterface $builder
+     * @param array                $options
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+
+        $leadChoices = array();
+
+        foreach ($options['data'] as $l) {
+             $leadChoices[$l->getId()] = $l->getPrimaryIdentifier();
+         }
+
+        $builder->add(
+            'lead_to_merge',
+            'choice',
+            array(
+                'choices'    => $leadChoices,
+                'label'      => $this->factory->getTranslator()->trans('mautic.lead.merge.select'),
+                'label_attr' => array('class' => 'control-label'),
+                'multiple'   => false,
+                'attr'       => array(
+                    'class' => 'form-control'
+                ),
+                'constraints' => array(
+                        new NotBlank(
+                            array(
+                                'message' => 'mautic.core.value.required'
+                            )
+                        )
+                    )
+            )
+        );
+
+        $builder->add('buttons', 'form_buttons', array(
+        'container_class' => 'lead-merge-buttons',
+        'apply_text'      => false,
+        'save_text'       => 'mautic.core.form.save'
+        ));
+
+        
+
+        
+        if (!empty($options["action"])) {
+            $builder->setAction($options["action"]);
+        }
+    }
+
+    /**
+     * @param OptionsResolverInterface $resolver
+     */
+    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    {
+        $resolver->setDefaults(
+            array(
+                'data_class'  => 'Mautic\LeadBundle\Entity\Lead',
```

Comment:
Need to remove this.  This tells Symfony that the underlying data is a Lead object leading to the error "The form's view data is expected to be an instance of class Mautic\LeadBundle\Entity\Lead, but is a(n) array. You can avoid this error by setting the "data_class" option to null or by adding a view transformer that transforms a(n) array to an instance of Mautic\LeadBundle\Entity\Lead. "